### PR TITLE
Update solver functions

### PR DESCRIPTION
**Pull request by pacospace**
- Created queries for solved/unsolved python packages
- Created queries for solved python packages with errors
- Updated schema for some inconsistency issue

### CODE REVIEW

**Comment by pacospace** [comment]
recheck

File: thoth/storages/graph/postgres.py [hunk 1]
```diff
@@ -489,67 +489,835 @@ def has_python_solver_error(
...
+        start_offset: int = 0,
+        count: int = _DEFAULT_COUNT,
+        package_name: str = None,
+        package_version: str = None,
+        index_url: str = None,
+        os_name: str = None,
+        os_version: str = None,
+        python_version: str = None,
+        distinct: bool = False,
+    ) -> List[Tuple[str, str, str]]:
```
Comment (fridex) [inline thoth/storages/graph/postgres.py]:
```python
    ) -> List[Tuple[str, Optional[str], Optional[str]]]:
```

File: thoth/storages/graph/postgres.py [hunk 2]
```diff
@@ -489,67 +489,835 @@ def has_python_solver_error(
...
+
+        if package_version is not None:
+            query = query.filter(PythonPackageVersionEntity.package_version == package_version)
+
+        query = query.offset(start_offset).limit(count/2)
+
+        if distinct:
+            query = query.distinct()
+
+        query = query.all()
```
Comment (fridex) [inline thoth/storages/graph/postgres.py]: Just rename `query` to result here:

```
        result = query.all()
```

As this code gets result, the return might be misleading.

File: thoth/storages/graph/postgres.py [hunk 3]
```diff
@@ -489,67 +489,835 @@ def has_python_solver_error(
...
+                    break
+
+        return query
+
+    def get_unsolved_python_package_edge_cases(
+        self,
+        *,
+        package_name: str = None,
+        package_version: str = None
+    ) -> List[Tuple[str, str, str]]:
```
Comment (fridex) [inline thoth/storages/graph/postgres.py]:
```
    ) -> List[Tuple[str, Optional[str], Optional[str]]]:
```

File: thoth/storages/graph/postgres.py [hunk 4]
```diff
@@ -489,67 +489,835 @@ def has_python_solver_error(
...
+            n = 1
+            for unsolved_tuple in unsolved:
+                query.append(unsolved_tuple)
+                n += 1
+                if n == count/2:
+                    break
+
+        return query
+
+    def get_unsolved_python_package_edge_cases(
```
Comment (fridex) [inline thoth/storages/graph/postgres.py]: I think this method could be private?

File: thoth/storages/graph/postgres.py [hunk 5]
```diff
@@ -593,85 +1361,260 @@ def retrieve_unanalyzed_python_package_versions(
...
+        count: int = _DEFAULT_COUNT,
+        package_name: str = None,
+        package_version: str = None,
+        index_url: str = None,
+        os_name: str = None,
+        os_version: str = None,
+        python_version: str = None,
+        distinct: bool = False,
+    ) -> List[Tuple[str, str, str]]:
+        """Retrieve solved Python package versions in Thoth Database.
```
Comment (fridex) [inline thoth/storages/graph/postgres.py]: With error?

File: thoth/storages/graph/postgres.py [hunk 6]
```diff
@@ -593,85 +1361,260 @@ def retrieve_unanalyzed_python_package_versions(
...
 
-        return (
-            self._session.query(PythonPackageVersion)
-            .join(Solved)
-            .filter_by(error=True, error_unsolvable=unsolvable, error_unparseable=unparseable)
-            .count()
-        )
+        return result
+
+    def get_error_solved_python_package_versions(
```
Comment (fridex) [inline thoth/storages/graph/postgres.py]: I wonder if we could merge this with `get_unsolvable_*` and `get_unparseable_*` methods and provide `unparseable`, `unsolvable` parameters to this method to query for the desired ones.

* `unsolvable=False, unparseable=False` - the current behavior of `get_error_solved_python_package_versions`
* `unsolvable=True, unparseable=False` - the current behavior of the unsolvable method
* `unsolvable=False, unparseable=True` - the current behavior of the unparseable method
* `unsolvable=True, unparseable=True` - raise an error here as this is not possible

File: thoth/storages/graph/postgres.py (hunk 1 above)
Comment (pacospace) [reply thoth/storages/graph/postgres.py]: Great, very nice, thanks :)

File: thoth/storages/graph/postgres.py (hunk 4 above)
Comment (pacospace) [reply thoth/storages/graph/postgres.py]: Yes right :) Thanks!

File: thoth/storages/graph/postgres.py (hunk 6 above)
Comment (pacospace) [reply thoth/storages/graph/postgres.py]:
```get_solved_python_package_versions_count_all()
69
get_error_solved_python_package_versions_count_all(unsolvable=True)
65
get_error_solved_python_package_versions_count_all(unparseable=True)
0
get_error_solved_python_package_versions_count_all()
67```

Comment (pacospace) [reply thoth/storages/graph/postgres.py]:
```
get_error_solved_python_package_versions_count_all(unsolvable=True, unparseable=True)
Traceback (most recent call last):
  File "query_test.py", line 667, in <module>
    result = graph.get_error_solved_python_package_versions_count_all(unsolvable=True, unparseable=True)
  File "/home/fmurdaca/work/aicoe/storages/thoth/storages/graph/postgres.py", line 1452, in get_error_solved_python_package_versions_count_all
    raise ValueError("Cannot query for unparseable and unsolvable at the same time")
ValueError: Cannot query for unparseable and unsolvable at the same time
```

Comment (fridex) [reply thoth/storages/graph/postgres.py]: Awesome! :+1:

**Pull request merged**